patches/server/0879-Add-DispenserReplaceItemEvent.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: xDec0de_ <63256529+xdec0de@users.noreply.github.com>
+Date: Thu, 3 Mar 2022 02:29:39 +0100
+Subject: [PATCH] Add DispenserReplaceItemEvent
+
+Currently handled for any bucket and bottle type, which should be all items that can be replaced on a dispenser as far as I know. This event could be used for servers with a rarity system on items (As mine), as currently I had to use the normal dispense event and delay 1 tick a loop through the dispenser inventory to change the item lore.
+
+diff --git a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+index 59db245fe11384282af84ec1d5be08ab0e9484ca..40beb96b083e716d5c349c96ee316f0d462122d5 100644
+--- a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
++++ b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+@@ -4,6 +4,8 @@ import java.util.Iterator;
+ import java.util.List;
+ import java.util.Optional;
+ import java.util.Random;
++
++import io.papermc.paper.event.block.DispenserReplaceItemEvent;
+ import net.minecraft.Util;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.BlockSource;
+@@ -632,6 +634,18 @@ public interface DispenseItemBehavior {
+                     dispensiblecontaineritem.checkExtraContent((Player) null, worldserver, stack, blockposition);
+                     // CraftBukkit start - Handle stacked buckets
+                     Item item = Items.BUCKET;
++
++                    // Paper start - DispenserReplaceItemEvent
++                    org.bukkit.block.Block block = worldserver.getWorld().getBlockAt(pointer.getPos().getX(), pointer.getPos().getY(), pointer.getPos().getZ());
++                    DispenserReplaceItemEvent replaceItemEvent = new DispenserReplaceItemEvent(block, stack.asBukkitCopy(), new ItemStack(item).asBukkitCopy());
++                    replaceItemEvent.callEvent();
++                    if (!replaceItemEvent.isCancelled()) {
```

Comment:
replaceItemIvent.callEvent() returns a boolean for cancellation - same thing for later usages